File: SampleProjects/codepilot/src/Class/Tables/RectangleTable.ts
import { Food } from "../Food";
import { FoodOnTable } from "../FoodOnTable";
import { ITable } from "../Interfaces/ITable";

export class RectangleTable implements ITable{
    x: number;
    y: number;
    isHovered: Boolean;
    tableId: number;
    isSelected: Boolean;
    color: String;
    tableStatus: String;
    shapeId: String;

    foods: FoodOnTable[];

    width: number;
    height: number;

    constructor(tableId: number, shapeId: String, x: number, y: number, width: number, height: number, tableStatus: String){
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.foods = [];
        this.tableStatus = tableStatus;
        this.tableId = tableId;
        this.shapeId = shapeId;
    }
    
    setIsHovered(value: boolean): void {
        this.isHovered = value;
        return;
    }
    
    isMouseInRange(x: number, y: number): boolean {
        return x > this.x && x < this.x + this.width && y > this.y && y < this.y + this.height
    }

    draw(context): void {
        if(this.tableStatus === "unlocked"){
            context.fillStyle = "#D9D9D9";
        }else if(this.tableStatus === "locked"){
            context.fillStyle = "#66DD56";
        }else if(this.tableStatus === "occupied"){
            context.fillStyle = "rgba(209, 150, 133, 0.69)";
        }else{
            throw Error("Not recognized behavior!");
        }

        context.fillRect(this.x, this.y, this.width, this.height);
        if(this.isSelected){
            context.strokeStyle = "rgba(123, 166, 180, 1)";
            context.lineWidth = 5;
            context.strokeRect(this.x, this.y, this.width, this.height);
        }
        else if(this.isHovered){
            context.strokeStyle = "rgba(173, 216, 230, 1)";
            context.lineWidth = 5;
            context.strokeRect(this.x, this.y, this.width, this.height);
        }
    }
}

File: SampleProjects/BlazorFE/BlazorApp1/Class/RectangleTable.cs
namespace BlazorApp1.Class
{
    using System;
    using System.Collections.Generic;

    public class RectangleTable : ITable
    {
        public int X { get; set; }
        public int Y { get; set; }
        public bool IsHovered { get; set; }
        public int TableId { get; set; }
        public bool IsSelected { get; set; }
        public string? Color { get; set; }
        public string? TableStatus { get; set; } // "locked", "unlocked", "occupied"
        public string? ShapeId { get; set; }
        public List<FoodOnTable> Foods { get; set; } = new List<FoodOnTable>();

        public int Width { get; set; }
        public int Height { get; set; }
        public string? ImgURL { get; set; }
        public string? ShapeName { get; set; }

        // Constructor
        public RectangleTable(int tableId, string shapeId, int x, int y, int width, int height, string tableStatus, string ImgURL, string ShapeName)
        {
            X = x;
            Y = y;
            Width = width;
            Height = height;
            Foods = new List<FoodOnTable>();
            TableStatus = tableStatus;
            TableId = tableId;
            ShapeId = shapeId;
            this.ImgURL = ImgURL;
            this.ShapeName = ShapeName;
        }

        public void SetIsHovered(bool value)
        {
            IsHovered = value;
        }

        public bool IsMouseInRange(int x, int y)
        {
            return x > X && x < X + Width && y > Y && y < Y + Height;
        }

        public void Draw(object context)
        {
            // Implement drawing logic using Blazor's Canvas or other graphics library
            // Placeholder for implementation
            throw new NotImplementedException();
        }
    }
}
